### DISCUSSION TITLE
Add security

### DISCUSSION THREAD

**Discussion by GqSteve** [roleoroleo/yi-hack-Allwinner#322] [Ideas]
Hi,

I raised an issue in the past and suggested to add MQTTS. I do understand that this is not so easy to develop.

Another option would be to add SSH and SCP in the firmware?! e.g. a SSH connection can be established from the camera to a remote host, equivalent SCP to copy files from the camera to a remote host.

How can I add SSH and SCP in the firmwares? I can make some tests and give feedback, how such a setup can work.

Thanks for the great work so far!

**Comment by roleoroleo** [comment]
> I raised an issue in the past and suggested to add MQTTS. I do understand that this is not so easy to develop.

I already tried to add "S" to MQTT but I encountered a lot or problems...

> Another option would be to add SSH and SCP in the firmware?! e.g. a SSH connection can be established from the camera to a remote host, equivalent SCP to copy files from the camera to a remote host.
> 
> How can I add SSH and SCP in the firmwares? I can make some tests and give feedback, how such a setup can work.
> 

SSH and SCP are already installed in the hack.

**Comment by GqSteve** [comment]
Indeed in /home/yi-hack/bin/.....then a link has to be added like 

ln -s /home/yi-hack/bin/ssh /home/base/ssh

My idea is to create a ssh connection to a remote ssh server and tunnel the 1883. 

Now the question is...can MQTT "run" when **only** this ssh connection is established? E.g. after a WLAN attack the camera will try to reconnect to the AP...as long as this ssh connection is not established, then MQTT should not post anything. What do you think?

**Reply by roleoroleo** [reply]
> Indeed in /home/yi-hack/bin/.....then a link has to be added like

It's not necessary. Load the env sourcing the file /home/yi-hack/script/env.sh

> My idea is to create a ssh connection to a remote ssh server and tunnel the 1883.
> 
> Now the question is...can MQTT "run" when **only** this ssh connection is established? E.g. after a WLAN attack the camera will try to reconnect to the AP...as long as this ssh connection is not established, then MQTT should not post anything. What do you think?

I don't understand, explain me better.
